_scripts/update.py

```diff
@@ -0,0 +1,179 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import tomli_w
+import tomllib
+from argparse import ArgumentParser
+from filecmp import cmp
+from os import makedirs
+from os.path import abspath, dirname, exists, join, relpath
+from re import match, sub
+from shutil import copy
+from sys import exit
+from compare_locales.merge import merge_channels
+from compare_locales.parser import Entity, getParser
+from compare_locales.paths import ProjectFiles, TOMLParser
+
+HEAD = "master"
+
+description = f"""
+Update the localization source files from a Firefox branch, adding new files and messages.
+For updates from the "{HEAD}" branch, also update changed messages.
+
+Writes a summary of the branch's localized files and message keys as `_data/[branch].json`,
+and a commit message summary as `.update_msg`.
+"""
+
+
+def add_config(fx_root: str, fx_cfg_path: str, done: set[str]):
+    parts = fx_cfg_path.split("/")
+    if (
+        "{" in fx_cfg_path
```

Comment:
Why do we check for `{` in the path to the TOML file?

Reply:
Yeah. This test passes for all `l10n.toml` files currently used in m-c, but it doesn't pass for all possible paths that may show up in `l10n.toml` files. Rather than pre-emptively building such support, let's make sure to fail here if they do show up later, and only then extend this.